charms/katib-db-manager/src/charm.py

```diff
@@ -165,123 +168,162 @@ def _on_mysql_relation(self, event):
         try:
             relation = self.model.get_relation("relational-db")
             if relation:
-                self.logger.warnning(
-                    "Failed to create mysql relation due to existing relational-db relation."
+                self.logger.warning(
+                    "Up-to-date database relation relational-db is already established."
+                )
+                self.logger.warning(f"{MYSQL_WARNING} {UNBLOCK_MESSAGE}")
+                self.model.unit.status = BlockedStatus(
```

Comment:
what is we fire later update status?

how do we get the same BlockedStatus ?

Reply:
is it line 380?
what if other event occurs? Will we enter this method as well?
```
    def _on_update_status(self, event):
        """Update status actions."""
        # skip update status processing in case of BlockedStatus
        if self.model.unit.status == BlockedStatus:
            return
```

Reply:
Re-writing it according ot Pietro's suggestion:
> you'd need an `isinstance(self.model.unit.status, BlockedStatus)`, but yeah.
That would mean you'd never recover from a blockedstatus though

[3h ago](https://chat.canonical.com/canonical/pl/u5gwsjesapf5ujzmdt9y5g7ano)
unless you set status from somewhere else as well, that is